lib/conversion.js

```diff
@@ -3,20 +3,58 @@ var streamMixins = require('./mixins/_stream');
 var readableMixins = require('./mixins/_readable');
 var writableMixins = require('./mixins/_writable');
 
+/**
+ * @class zstreams
+ * @static
+ * @method isClassicStream
+ * @param {Stream} stream - Stream to check
+ * @return {Boolean} - Whether or not the stream is a "classic" stream
+ */
+function isClassicStream(stream) {
+	return !(stream.pipe && (stream.read || (stream.write && !(stream.destroy && stream.end))));
+}
+exports.isClassicStream = isClassicStream;
+
+/**
+ * @class zstreams
+ * @static
+ * @method isRequestStream
+ * @param {Stream} stream - Stream to check
+ * @return {Boolean} - Whether or not the stream is a "request" stream
+ */
 function isRequestStream(stream) {
-	return !!(stream.pipe && !stream.read && stream.headers);
+	return !!(isClassicStream(stream) && !!stream.headers && !!stream.method);
```

Comment:
Why all the !!s?

return !!(isClassicStream(stream) && stream.headers && stream.method);